internal/neofs/neofs.go

```diff
@@ -278,6 +270,32 @@ func (x *NeoFS) CreateObject(ctx context.Context, prm layer.PrmObjectCreate) (oi
 		return oid.ID{}, fmt.Errorf("save object via connection pool: %w", err)
 	}
 
+	bChunk := make([]byte, ni.MaxObjectSize())
+	var n int
+	var idObj oid.ID
+
+	for {
```

Comment:
The same question about `io.Copy` is relevant here as well.

Reply:
Updated to the same solution as in HTTP gate